File: apis/src/main/java/com/beat/apis/schedule/application/dto/response/TicketAvailabilityResponse.java
package com.beat.apis.schedule.application.dto.response;

public record TicketAvailabilityResponse(
	Long scheduleId,
	String scheduleNumber,
	int totalTicketCount,
	int soldTicketCount,
	int availableTicketCount,
	int requestedTicketCount,
	boolean isAvailable
) {
	public static TicketAvailabilityResponse of(Long scheduleId, String scheduleNumber, int totalTicketCount,
		int soldTicketCount, int availableTicketCount, int requestedTicketCount, boolean isAvailable) {
		return new TicketAvailabilityResponse(scheduleId, scheduleNumber, totalTicketCount, soldTicketCount,
			availableTicketCount, requestedTicketCount, isAvailable);
	}
}

File: apis/src/main/kotlin/com/beat/apis/schedule/api/response/TicketAvailabilityResponse.kt
package com.beat.apis.schedule.api.response

import com.beat.apis.schedule.application.result.TicketAvailabilityResult
import com.fasterxml.jackson.annotation.JsonProperty

@ConsistentCopyVisibility
data class TicketAvailabilityResponse private constructor(
    val scheduleId: Long?,
    val scheduleNumber: String?,
    val totalTicketCount: Int,
    val soldTicketCount: Int,
    val availableTicketCount: Int,
    val requestedTicketCount: Int,
    @get:JsonProperty("isAvailable")
    val isAvailable: Boolean,
) {
    companion object {
        fun from(result: TicketAvailabilityResult): TicketAvailabilityResponse = TicketAvailabilityResponse(
            scheduleId = result.scheduleId,
            scheduleNumber = result.scheduleNumber,
            totalTicketCount = result.totalTicketCount,
            soldTicketCount = result.soldTicketCount,
            availableTicketCount = result.availableTicketCount,
            requestedTicketCount = result.requestedTicketCount,
            isAvailable = result.isAvailable,
        )
    }
}
